### ISSUE TITLE
Final report peer review

### ISSUE THREAD

**Issue by s0ap** [hya-cala/PredictH1BVisa#14]
The project is to analyze and predict the outcome of the LCA which is filed by employers to be considered for the H1-B lottery. The dataset used contains relevant information about a foreign employee in a company such as salary, job type, industry classification of the company, etc. This is a good problem to analyze since it can help both employer and employee plan better.

Things I like:
- The business problem and its impact have been very well detailed and show the importance of predicting the application status.
- The exploratory data analysis section is impressive with the use of violin plots, map plots along with histograms, and bar charts to explain the feature distributions in detail. They have correctly applied feature engineering as well considering missing value treatment and transforming/normalizing data for the model input.
- I really liked the approach of first checking results from an AutoML package - and then narrowing down the models to use. All the models in the report have detailed mathematical explanations as well.
- They have correctly identified the imbalance in the dataset and worked towards minimizing its impact on the final results.

Things to improve:
- Some of the key features from the dataset have been dropped such as sector and year - which I think are important since the LCA approval process may have changed from year to year and a lot of H1B applications tend to be skewed towards the tech and finance/banking sectors.
- They have tried out a lot of models for the project - but some of them are missing details on why the model did not perform well. Ideally, they should have added some explanation on why the particular model did not work.
- They have not accounted for overfitting which is especially important when using non-linear models. Also, they did not specify an explicit validation set - or use cross-validation for hyperparameter tuning.
- They have used the accuracy score (balanced accuracy) for reporting the results - additional metrics should be used to evaluate the results too (I noticed they used F1-score in their midterm report - why was it not reported here?).
